Start test debug session in aggregate project

*Copy of https://github.com/scalameta/metals/issues/2690 reported by @markusa380*

When
* the build.sbt contains two projects
* one project ("root") contains a testing framework and tests, and `.aggregate`s the other project ("common")
* SBT is used as language server

running the `test` decorator or the `metals.test-current-file` fails with an ugly error message in the Metals output:

```
...
[Error - 11:36:15 AM] Request workspace/executeCommand failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  no matching test group
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  no matching test group
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:237)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:229)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:417)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
	... 4 more

2021.04.08 11:36:14 ERROR no matching test group
2021.04.08 11:36:14 ERROR (common / Test / startTestSuitesDebugSession) no matching test group
2021.04.08 11:36:14 ERROR Error -33000: 
...
```

**To Reproduce**

Steps to reproduce the behavior:

1. Clone this repository: https://github.com/markusa380/metals-test-error
2. Select SBT as build server (didn't test with Bloop)
3. Run the `test` decorator in `HelloSpec.scala`

**Expected behavior**

Tests work.

**Installation:**

- Operating system: macOS
- Editor: Visual Studio Code
- Metals version: v0.10.1

**Additional context**

Only tested with SBT as build server.

**Search terms**

test, decorator, command, sbt

--- a/sbt/plugin/src/main/scala/ch/epfl/scala/debugadapter/sbtplugin/DebugAdapterPlugin.scala
+++ b/sbt/plugin/src/main/scala/ch/epfl/scala/debugadapter/sbtplugin/DebugAdapterPlugin.scala
@@ -69,9 +69,14 @@ object DebugAdapterPlugin extends sbt.AutoPlugin {
     }
   )
 
-  override def projectSettings: Seq[Def.Setting[_]] = {
-    inConfig(Compile)(runSettings) ++ inConfig(Test)(testSettings)
-  }
+  override def projectSettings: Seq[Def.Setting[_]] = Def.settings(
+    inConfig(Compile)(runSettings),
+    inConfig(Test)(testSettings),
+    startMainClassDebugSession / Keys.aggregate := false,
+    startTestSuitesDebugSession / Keys.aggregate := false,
+    startRemoteDebugSession / Keys.aggregate := false,
+    stopDebugSession / Keys.aggregate := false
+  )
 
   def runSettings: Seq[Def.Setting[_]] = Seq(
     startMainClassDebugSession := mainClassSessionTask.evaluated,

--- a/sbt/plugin/src/sbt-test/debug-session/aggregate/build.sbt
+++ b/sbt/plugin/src/sbt-test/debug-session/aggregate/build.sbt
@@ -1,0 +1,9 @@
+ThisBuild / scalaVersion := "2.13.4"
+
+lazy val common = project.in(file("common"))
+
+lazy val root = project.in(file("."))
+  .aggregate(common)
+  .settings(
+    libraryDependencies += "org.scalatest" %% "scalatest" % "3.2.2",
+  )

--- a/sbt/plugin/src/sbt-test/debug-session/aggregate/project/plugins.sbt
+++ b/sbt/plugin/src/sbt-test/debug-session/aggregate/project/plugins.sbt
@@ -1,0 +1,11 @@
+val pluginVersion = sys.props.get("plugin.version")
+  .getOrElse {
+    sys.error(
+      """|The system property 'plugin.version' is not defined.
+         |Specify this property using the scriptedLaunchOpts -D.
+         |""".stripMargin
+    )
+  }
+
+addSbtPlugin("ch.epfl.scala" % "sbt-debug-adapter" % pluginVersion)
+libraryDependencies += "ch.epfl.scala" %% "debug-adapter-test-client" % pluginVersion

--- a/sbt/plugin/src/sbt-test/debug-session/aggregate/project/project/plugins.sbt
+++ b/sbt/plugin/src/sbt-test/debug-session/aggregate/project/project/plugins.sbt
@@ -1,0 +1,2 @@
+// required for adopt@1.8
+addSbtPlugin("org.scala-debugger" % "sbt-jdi-tools" % "1.1.1")

--- a/sbt/plugin/src/sbt-test/debug-session/aggregate/src/main/scala/example/Hello.scala
+++ b/sbt/plugin/src/sbt-test/debug-session/aggregate/src/main/scala/example/Hello.scala
@@ -1,0 +1,9 @@
+package example
+
+object Hello extends Greeting with App {
+  println(greeting)
+}
+
+trait Greeting {
+  lazy val greeting: String = "hello"
+}

--- a/sbt/plugin/src/sbt-test/debug-session/aggregate/src/test/scala/example/HelloSpec.scala
+++ b/sbt/plugin/src/sbt-test/debug-session/aggregate/src/test/scala/example/HelloSpec.scala
@@ -1,0 +1,10 @@
+package example
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class HelloSpec extends AnyFlatSpec with Matchers {
+  "The Hello object" should "say hello" in {
+    Hello.greeting shouldEqual "hello"
+  }
+}

--- a/sbt/plugin/src/sbt-test/debug-session/aggregate/test
+++ b/sbt/plugin/src/sbt-test/debug-session/aggregate/test
@@ -1,0 +1,3 @@
+> 'root / Test / startTestSuitesDebugSession ["example.HelloSpec"]'
+> 'root / Test / startMainClassDebugSession {"class":"example.Hello","arguments":[],"jvmOptions":[]}'
+> root / Test / stopDebugSession

